[1.13.2] Add getter method for MouseHelper.middleDown

Geforce132 opened the pull request:
The current MouseHelper class has getters for leftDown and rightDown, which are booleans that return true if the player is holding down their left or right mouse buttons respectively. However, for some reason, MouseHelper doesn't have a getter method for middleDown, which is a boolean that checks whether or not the player's middle mouse button is being held down. This could be a useful method for mods that want to utilize middle-mouse to perform some function (including one of mine). 

An AccessTransformer or Reflection could also allow us access to this private value, but that's fairly impractical to have to set up an AT or deal with the performance issues of Reflection (if you're referencing this boolean a lot) when there's already getters for the other two mouse buttons. This commit simply adds a return method for middleDown into MouseHelper.

bs2609 committed:
Make Forge-provided default transforms accessible to custom models

KidsDontPlay committed:
fixed visual bug with guislider

bs2609 committed:
Make Forge blockstate variants correctly inherit AO setting from vanilla models (#5190)

bs2609 committed:
Fix some null returns from defaulted registries (#5235)

bs2609 committed:
Fix Baked Item models with transformations (#5241)

KidsDontPlay committed:
Fix an ObjectHolderRef internal error message (#5214)

tterrag1098 committed:
Allow loading json constants outside of _constants

ichttt committed:
Add a hook for farmland watering (#4891)

mcenderdragon committed:
added PotionHooks, closes #3867, #4375 (#4614)

tterrag1098 committed:
Fix potion remove event not always firing, add expiry event

LexManos committed:
Fix potential issues with the Minecraft FakePlayer lingering around after world unloads.

bs2609 committed:
Ensure slave maps are cleaned up when handling registry overrides (#5250)

bs2609 committed:
Fix small logic error in emissive item rendering code (#5320)

bs2609 committed:
Add redirects to PotionEffect to respect registry replacement (#5213)

tterrag1098 committed:
Support diffuse lighting flag in item rendering

tterrag1098 committed:
Simplify custom item rendering by removing GL emissivity hacks

bs2609 committed:
Fix small logic error in ItemTextureQuadConverter (#5463)

Cadiboo committed:
Small comment typo fixes for ObfuscationReflectionHelper (#5583)

coaster3000 committed:
Add EntityPlaceEvent (#5057)

Co-authored-by: tterrag <tterrag1098@gmail.com>

bs2609 committed:
Only prompt for missing registries on local worlds (#5348)

Darkhax committed:
Fix special spawn event not firing in many cases. (#5389)

Co-authored-by: tterrag <tterrag1098@gmail.com>

Landmaster committed:
Add a few events pertaining to villages (#5302)

tterrag1098 committed:
Exclude noci branches from travis

tterrag1098 committed:
Exclude PR builds from discord and changelog

Geforce132 committed:
[1.13.2] Add getter method for MouseHelper.middleDown

williewillus commented:
Is there any other private data in this class we want to expose too? So we can get em all at once

Geforce132 commented:
I don't believe so? The only other private variables that I think would ever be used by any type of mod in MouseHelper is x/yVelocity, but I don't know why that would be needed.

Wyn-Price committed:
Added more Constants (#5323)

cpw committed:
Allow login packets to only generate on server based connections
Run injectSnapshot on the client thread, wait for result and
disconnect when injectSnapshot fails.
Fix DevServer to load MOD_CLASSES
Fix LaunchTesting to properly load log4j2 markers (don't use a
static variable in that class!)

Signed-off-by: cpw <cpw+github@weeksfamily.ca>

tterrag1098 committed:
Handle null return from ResourcePackInfo.createResourcePack

bs2609 committed:
Fix getDefaultCreatorModId for enchanted books (#5534)

JoJoDeveloping committed:
Re-Add the list ping compatibility checker

Signed-off-by: JoJoDeveloping <jojohostert@gmail.com>

LexManos committed:
Fix NPE in TileEntityPistonRender when class is created to early.

LexManos committed:
Do not invalidate caps when entity is being moved across worlds. Should fix #5500

LexManos committed:
Revive CapabilityProvider on entity revive. Madders should NOT need this, as caps are copied via NBT.

Hugman76 committed:
GEMS_QUARRTZ -> GEMS_QUARTZ (#5635)

Pokechu22 committed:
Fail fast when null is used with setTag instead of crashing in writeEntry (#5257)

tterrag1098 committed:
Comment out test Mod/EventBusSubscriber annotations, fix up ModelLoaderRegistryTest

tterrag1098 committed:
Fix loading errors GUI crashing when errors occur during construction

tterrag1098 committed:
Update Night-Config to 3.5.0, improve defineEnum methods

Closes #5537

Cadiboo commented:
> The only other private variables that I think would ever be used by any type of mod in MouseHelper is x/yVelocity, but I don't know why that would be needed.

I can think of mods would render something (maybe an entity or model) looking in direction and have it moveable with click & drag, which will probably use velocity

williewillus committed:
Fix PlayerInteractEvent using Dist when it should be using LogicalSide (#5611)

mezz committed:
Remove uses of java.awt (#5591)

`java.awt` will statically load itself when `java.awt.Color`, `java.awt.Dimension`, etc are loaded.
This conflicts with lwjgl3 and must be avoided (except for in mojang's server gui).

Shadows-of-Fire committed:
Add class to wrap IItemHandler as IInventory for easier recipe interaction. (#5522)

cpw committed:
Defer configuration of channel until it is actually available during
connection. Closes #5626

Signed-off-by: cpw <cpw+github@weeksfamily.ca>

JoJoDeveloping committed:
Remove game registry comparsion from list ping

Signed-off-by: JoJoDeveloping <jojohostert@gmail.com>

tterrag1098 committed:
Add method to IResourcePack to hide it entirely from the UI (#5525)

cpw committed:
Remove redundant method call in Minecraft patch. Also fix a translation
string error (closes #5643)

Signed-off-by: cpw <cpw+github@weeksfamily.ca>

cpw committed:
Fire the ModIdRemappingEvent on the Forge Event bus. Closes #5632

Signed-off-by: cpw <cpw+github@weeksfamily.ca>

tterrag1098 committed:
Add back EntityPlayer patch to return main inventory in IItemHandler

Fixes #5638

tterrag1098 committed:
Fix BrandingControl caching text through resource reload

tterrag1098 committed:
Remove remaining deprecations, improve a few patch comments

cpw committed:
Fix StartupQuery to run properly on dedicated server.. Closes #5649

Signed-off-by: cpw <cpw+github@weeksfamily.ca>

cpw committed:
Merge branch 'pull/5636' into 1.13.x

cpw committed:
Implement API for mods to control their server status response..

Signed-off-by: cpw <cpw+github@weeksfamily.ca>

williewillus committed:
Fix respawnevent being fired with old player instead of new. Fixes #5658 (#5668)

cpw committed:
Refactor to use naming service, and provide an MCP naming service when
in userdev. Step 1 to supporting loading SRG mods in userdev.

Signed-off-by: cpw <cpw+github@weeksfamily.ca>

cpw committed:
Actually parse the i18n exception message into the super constructor,
so the exception has a useful text message, even in crash logs and
on the server.

Signed-off-by: cpw <cpw+github@weeksfamily.ca>

ben-mkiv committed:
FENCES_NETHER_BRRICK => FENCES_NETHER_BRICK (#5681)

Bricktricker committed:
dimesnion -> dimension typos (#5684)

bs2609 committed:
Allow custom DataSerializers to be registered safely (#5245)

bs2609 committed:
Catch json parsing errors from constants/factories files (#5258)

ichttt committed:
Minor performace improvement when building chunks and rendering blocks (#5286)

bs2609 committed:
Add Forge dimension-changing hooks to spectator handling code (#5212)

DarkGuardsman committed:
Fix for SidedInvWrapper isItemValid using wrong slot (#5642)

bs2609 committed:
Transform vertex normals as well as positions when generating quads (#5242)

bs2609 committed:
Fix some vertex format changes not being handled correctly (#5368)

tterrag1098 committed:
Update skipped_commits.md

ustc-zzzz committed:
Only remove synthetic lambda methods referenced in body of SideOnly methods (#5127)

TheIllusiveC4 committed:
Fix the bed position given to the SleepingTimeCheck event (#5107)

bs2609 committed:
Re-add some missing villager profession patches (#5200)

* Fix zombie villagers only spawning with vanilla professions
* Fix spawning modded villagers that do not have their own building

bs2609 committed:
Fix missing string parameters in some log messages (#5210)

williewillus committed:
Stop firing LivingSetAttackTargetEvent for setRevengeTarget (#5217)

tterrag1098 committed:
Update skipped_commits.md

pau101 committed:
Fix inaccurate main thread name shown in client log (#5078)

tterrag1098 committed:
Implement rendering for item models with emissive quads (#5047)

bs2609 committed:
Improve context provided by potion icon rendering hooks (#5111)

bs2609 committed:
Improve exception handling from server starting events (#5226)

tterrag1098 committed:
Update skipped_commits.md

bs2609 committed:
Allow items to control the rate of repair from mending (#5102)

bs2609 committed:
Improve support for custom block path types (#5203)

Co-authored-by: tterrag <tterrag1098@gmail.com>

tterrag1098 committed:
Fix #5659 setdimension command entity filtering, avoid inverting logic

ichttt committed:
Add logging for data manager key registration errors (#5129)

tterrag1098 committed:
Update skipped_commits.md

tterrag1098 committed:
Fix some typos in cherry-picked commits

tterrag1098 committed:
Fix EntityDataManager logging level

tterrag1098 committed:
Update skipped_commits.md

tterrag1098 committed:
Finish model loader debug mod

tterrag1098 committed:
Fix compile error in FakePlayer from cherry pick

0uti committed:
[1.13.x] Fire InputEvents (#5533)

Larsg310 committed:
Add GuiContainerEvent.DrawBackground event. (#5595)

DaemonUmbra committed:
Remove Vanilla Logging Config (#5618)

This will re-enable Forge's logging configuration in "production"

williewillus committed:
Fix some typos in spawn dimension patches (#5669)

tterrag1098 committed:
Fix #5688 config values are unordered, update Night-Config

tterrag1098 committed:
Remove hack to force enum configs to strings

ichttt committed:
Readd redirecting of stdout and stderr (#5678)

ichttt committed:
Cleanup/Implement some todos (#5660)

tterrag1098 committed:
Add FieldRedirectTransformer updated to new modlauncher API

tterrag1098 committed:
Update skipped_commits.md

bs2609 committed:
Add a hook to allow continuously using items through stack changes (#4834)

LexManos committed:
Extend Region files to support >1MB per chunk.

If the 'sector count' is 255, ask the compressed data header for the proper length.

LexManos committed:
Written size does not include int bytes.

tterrag1098 committed:
Update skipped_commits.md

bs2609 committed:
Fix up torch placement logic to handle more vanilla special-casing (#5426)

bs2609 committed:
Better support for custom bows (#5209)

tterrag1098 committed:
Reimplement ForgeHooks#onPlaceItemIntoWorld and remove old place events

tterrag1098 committed:
Update skipped_commits.md

tterrag1098 committed:
Sync up canPlaceTorchOnTop logic with BlockTorch#isValidPosition

tterrag1098 committed:
Update Constants for 1.13

tterrag1098 committed:
Update skipped_commits.md

tterrag1098 committed:
Remove unnecessary patch

tterrag1098 committed:
Further clean up Constants, and add two new BlockFlags

tterrag1098 committed:
Fix MC-128441: cross-dimensional teleporting does not work

tterrag1098 committed:
Change default username to 'Dev', add ability to replace '#' with randomized numbers

tterrag1098 committed:
Add comment to vanilla bugfix, update skipped_commits.md

tterrag1098 committed:
Update skipped_commits.md

tterrag1098 committed:
Make EnumRarity extensible

tterrag1098 committed:
Fix EnumRarity, make sure extensible enums have at least one factory candidate

tterrag1098 committed:
Add explicit return type checks for extensible enums, fix exc

LexManos committed:
Bump ForgeSPI version, and add support for DistCleaner stripping interfaces from classes.
Add more detail to VersionChecker log info.

tterrag1098 committed:
Add new system for model data, replacing extended states (#5564)

OhYea777 committed:
Update 'build.gradle' to fix run configs for launching forge

tterrag1098 committed:
Better handle errors from eclipse compiled (non-compiling) bytecode

tterrag1098 committed:
Add test mods run configs

tterrag1098 committed:
Fix BakedQuadRetextured calling old BakedQuad ctor

cpw committed:
Updated a couple of libraries. Fixed bug where modid could be in toml but
not found during loading, yet game continued to load.

Signed-off-by: cpw <cpw+github@weeksfamily.ca>

tterrag1098 committed:
Add ModelDataTest to test mods.toml

tterrag1098 committed:
Fix default IUnbakedModel bake method using wrong format

tterrag1098 committed:
Remove patch causing TEs to be removed on state change

cpw committed:
Fix #5735 by pushing new build. Shutdown Forge eventbus when errors occur. Closes #5711
Also, improve logging so warnings, errors and fatals are always shown, regardless of marker.

Signed-off-by: cpw <cpw+github@weeksfamily.ca>

tterrag1098 committed:
Fix modded TEs not being removed on block change

williewillus committed:
Add support for vanilla recipes having nbt on result (#5710) Fix #5663

LexManos committed:
Fix vanilla bug in writing structure templates with multiple palettes.

williewillus committed:
Fix caps not being visible on old player during clone event (#5728)

LexManos committed:
Fix build script producing duplicated eclipse source folders.
Fix not including userdev source in sources jar.
Update license header of files that need it.
Delete patches_old folders. Most are unneeded in 1.13+ as the systems they effected were changed significantly. Any that need to be readdressed can be done as PRs.

mezz committed:
Fix MC-146650 GuiContainer#keyPressed to return the correct key handled boolean (#5367)

bl4ckscor3 commented:
Merging this would be very helpful for ease of accessibility.

tterrag1098 committed:
Remove cache eviction for model data, optimize

cpw committed:
Enhance crash reports and logging to capture transformers on stack trace elements.

Moved log4j2.xml file to the launcher source set, so it loads first in the client.

Signed-off-by: cpw <cpw+github@weeksfamily.ca>

LexManos committed:
Fix tall grass not dropping seeds. Closes #5675

LexManos committed:
Fix vanilla structures loosing valid flags, And fix error placement for unsaveable structures.

LexManos committed:
Include new library in installer json.
Change order of versions on main menu, to prevent overlap with default window size.
Fix Forge update not displaying version info.

mezz commented on patches/minecraft/net/minecraft/client/MouseHelper.java.patch:
```diff
@@ -40,3 +40,14 @@
+       return this.field_198039_d;
+    }
+ 
++   public boolean isMiddleDown() {
```
Please put new methods at the bottom of the file, after a comment like this:
` /* ======================================== FORGE START =====================================*/`

mezz commented:
Superseded by https://github.com/MinecraftForge/MinecraftForge/pull/5776
